### PR TITLE
[Hexagon] Adjust RPC read buffer size from python 

### PR DESCRIPTION
**Pull request by mehrdadh**
This PR changes RPC read buffer size to be adjustable from python size by registering `init` function for FastRPC.

@csullivan @kparzysz-quic

### CODE REVIEW

File: src/runtime/hexagon/rpc/hexagon/rpc_server.cc [hunk 1]
```diff
@@ -196,10 +195,16 @@ class HexagonRPCServer {
 }  // namespace tvm
 
 namespace {
-tvm::runtime::hexagon::HexagonRPCServer* get_hexagon_rpc_server() {
-  static tvm::runtime::hexagon::HexagonRPCServer g_hexagon_rpc_server(
-      new uint8_t[TVM_HEXAGON_RPC_BUFF_SIZE_BYTES], TVM_HEXAGON_RPC_BUFF_SIZE_BYTES);
-  return &g_hexagon_rpc_server;
+static tvm::runtime::hexagon::HexagonRPCServer* g_hexagon_rpc_server;
+tvm::runtime::hexagon::HexagonRPCServer* get_hexagon_rpc_server(
+    uint32_t rpc_receive_buff_size_bytes = TVM_HEXAGON_RPC_BUFF_DEFAULT_SIZE_BYTES) {
```
Comment (csullivan) [inline src/runtime/hexagon/rpc/hexagon/rpc_server.cc:200]: How about we set the default value to 0, delete the TVM_HEXAGON_RPC_BUFF_DEFAULT_SIZE_BYTES define, and then CHECK that `rpc_receive_buff_size_bytes` is non-zero when `g_hexagon_rpc_server` is null?

File: src/runtime/hexagon/rpc/simulator/session.cc [hunk 2]
```diff
@@ -1311,6 +1311,8 @@ detail::Optional<HEXAPI_Nullptr> SimulatorRPCChannel::to_nullptr(const detail::M
 
 TVM_REGISTER_GLOBAL("tvm.contrib.hexagon.create_hexagon_session")
```
Comment (csullivan) [inline src/runtime/hexagon/rpc/simulator/session.cc:1312]: Can we include a test that we expect to fail when the stack size is too small, and expect to pass when the stack size is set larger?

**Review by csullivan — CHANGES_REQUESTED** [review]
🎉

File: src/runtime/hexagon/rpc/simulator/session.cc (hunk 2 above)
Comment (kparzysz-quic) [reply src/runtime/hexagon/rpc/simulator/session.cc:1312]: The simulator ignores the stack size at the moment.

File: src/runtime/hexagon/rpc/hexagon/rpc_server.cc (hunk 1 above)
Comment (mehrdadh) [reply src/runtime/hexagon/rpc/hexagon/rpc_server.cc]: done.

**Review by csullivan — APPROVED** [review]
LGTM!

File: src/runtime/hexagon/rpc/simulator/session.cc (hunk 2 above)
Comment (kparzysz-quic) [reply src/runtime/hexagon/rpc/simulator/session.cc:1312]: https://github.com/apache/tvm/pull/11046

**Comment by kparzysz-quic** [comment]
Is there a reason not to use `std::stringbuf` instead of the preallocated array? (the code for simulator does that)

**Comment by mehrdadh** [comment]
@kparzysz-quic I don't see a reason. Would std::stringbuf be slower?

**Comment by kparzysz-quic** [comment]
> @kparzysz-quic I don't see a reason. Would std::stringbuf be slower?

Probably, but it wouldn't affect model performance, since the only time it would run would be during the actual RPC communication.  If you want you can consider it in another PR.

**Review by kparzysz-quic — APPROVED** [review]

**Pull request merged**